### PR TITLE
Feature/validate and deploy IG

### PR DESCRIPTION
**Pull request by YanLabCHU**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+          
+      - name: Run script
+        env:
+          REPO_TOKEN: ${{ secrets.CLIN_FHIR_DEPLOY_KEY }}
+        run: |
+          SCRIPT=.github/build.sh
+          if [ -f $SCRIPT ]; then 
+            chmod u+x $SCRIPT
+            $SCRIPT
+          fi
```
Comment (Magnitus-) [inline .github/workflows/build.yml:25]: I would remove the 'if' condition here and just run the script.

If the script is missing, it should not silently do nothing. Rather, the pipeline should be allowed to fail as something is clearly wrong (either the script should be re-added or the pipeline should also be deleted at that point)

File: .github/build.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+echo "Running pip install"
+pip install .
+
+sudo mv site_root/input/resources/terminology/CodeSystem-hp.json CodeSystem-hp.json
+fhirutil validate --publisher_opts="n/a" --clear_output site_root/ig.ini
+sudo mv CodeSystem-hp.json site_root/input/resources/terminology/CodeSystem-hp.json
```
Comment (Magnitus-) [inline .github/build.sh:9]: Its a little bit weird that you need sudo for the mv commands here.

Intuitively, I would expect the running user here to be the same user that was used to clone the repo. Have you tried it with the sudo?

File: .github/build.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+# replace gh-pages root ig directory with site_root/output
+git remote set-branches origin '*'
+git fetch origin gh-pages
+git checkout -f gh-pages
+git rm -rf --ignore-unmatch ig
+sudo mv -f site_root/output ig
+sudo mkdir -p ignored
+sudo mv ig/*.zip ignored
+sudo mv ig/*.tgz ignored
+sudo mv ig/*.pack ignored
```
Comment (Magnitus-) [inline .github/build.sh:20]: Similar comment here. Do you need the sudo?

Comment (Magnitus-) [reply .github/build.sh:20]: Also, I would put a space and a comment here.

There should probably be a clear separation between the file manipulations above and the ensuing git manipulations that will commit the change and push it on the repo

Comment (YanLabCHU) [reply .github/build.sh:20]: Yes, we need it here.

**Review by Magnitus- — APPROVED** [review]
LGTM

**Pull request merged**